pkg/api/aim/encoding/decoder.go

```diff
@@ -42,36 +43,70 @@ func (d *reader) readField() ([]byte, error) {
 	return data, nil
 }
 
-// Decode decodes input stream of data into map[string]interface{}.
+// DecoderProvider provides an interface to work with stream decoder.
+type DecoderProvider interface {
+	// Next implements iterator pattern and return data set by set.
+	Next() (map[string]interface{}, error)
+	// Decode decodes all the data stream at once.
+	Decode() (map[string]interface{}, error)
+}
+
+// Decoder represents decoders for streaming data.
+type Decoder struct {
+	path     []string
+	reader   reader
+	cursor   string
+	position int
```

Comment:
this does not seem to be used anymore

Reply:
yep, left it for debugging purpose but I can remove it.